src/test/kotlin/ai/blindspot/ktoolz/extensions/OtherExtensionsTest.kt

```diff
@@ -151,4 +151,30 @@ internal class OtherExtensionsTest {
         }, block = { set(0, 0) }))
         assertEquals(1, listUnderTest[0])
     }
+
+    @Test
+    fun restrictLengthWithEllipsis() {
+        val result = "ABCDEFGHIJK".restrictLengthWithEllipsis(8, "elip")
+        assertEquals("ABCDelip", result)
+    }
+
+    @Test
+    fun toLongDebugString() {
+        assertEquals("Double(42.0)", 42.0.toLongDebugString())
+        assertEquals("MyClass[short]", "Any object".toLongDebugString("short","[]", className = "MyClass"))
+    }
+
+    @Test
+    fun toShortDebugString() {
+        val short = "SHORT"
+        val long = short.toLongDebugString()
+        assertEquals( short, long.toShortDebugString())
+    }
+
+    @Test
+    fun itemsToDebugString() {
+        val itemToString : (Int)->String = {i -> "NUM$i"}
+        val result = setOf(10, 20).itemsToDebugString("numbers", itemToString = itemToString )
+        assertEquals("2 numbers: NUM10, NUM20", result)
```

Comment:
This is a very strong assertion, sensitive for micro changes in format. But I think it is better than the alternatives.

Reply:
Microassertions are actually fine in my opinion. That's because you're testing something that you know how it should behave, when there's even little deviation in the output, test should fail.

Reply:
What is the difference between assertion and microassertion? Never heard the later and didn't find it on google...